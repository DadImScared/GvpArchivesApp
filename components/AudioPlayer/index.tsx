import * as React from "react";
import { StyleSheet, View } from "react-native";

import { withAudioPlayer } from "../withAudioPlayer";
import ButtonGroup from "./ButtonGroup";
import DurationSeeker from "./DurationSeeker";
import { IAudioPlayerState } from "./Provider";
import Title from "./Title";

interface IProps {
  audio: IAudioPlayerState;
}

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: "white",
    bottom: 0,
    padding: 8,
    position: "absolute",
    width: "100%",
  }
});

export class AudioPlayer extends React.Component<IProps> {
  render() {
    const {
      audio: {
        isSeeking,
        isPlaying,
        onPlayPausePressed,
        getSongName,
        getSeekSliderPosition,
        getTimeStamp,
        onSeekSliderChange,
        onSeekSliderComplete

      }
    } = this.props;
    return (
      <View style={styles.wrapper}>
        <Title getSongName={getSongName} />
        <DurationSeeker
          isSeeking={isSeeking}
          getSeekSliderPosition={getSeekSliderPosition}
          getTimeStamp={getTimeStamp}
          onSeekSliderChange={onSeekSliderChange}
          onSeekSliderComplete={onSeekSliderComplete}
        />
        <ButtonGroup
          onPlayPausePressed={onPlayPausePressed}
          isPlaying={isPlaying}
        />
      </View>
    );
  }
}

export default withAudioPlayer(AudioPlayer) as any; // property error without "any"
