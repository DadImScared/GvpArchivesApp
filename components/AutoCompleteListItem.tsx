import * as React from "react";
import { Text } from "react-native";

import { ListItem } from "native-base";

interface IProps {
  text: string;
}

export class AutoCompleteListItem extends React.Component<IProps> {

  render() {
    return (
      <ListItem button={true} noIndent={true}>
        <Text>{this.props.text}</Text>
      </ListItem>
    );
  }
}

export default AutoCompleteListItem;
