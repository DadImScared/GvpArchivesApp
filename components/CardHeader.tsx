import * as React from "react";

import { Body, CardItem, Text } from "native-base";
import { routes } from "./SideBar";

interface IProps {
  item: any;
  extraSubHeader?: string;
}

export const CardHeader: React.FunctionComponent<IProps> = ({ item, extraSubHeader }) => (
  <CardItem header={true}>
    <Body>
      <Text>{item.title.trim()}</Text>
      <Text note={true}>{routes[item.category]}{extraSubHeader ? `${extraSubHeader}` : null}</Text>
    </Body>
  </CardItem>
);

export default CardHeader;
