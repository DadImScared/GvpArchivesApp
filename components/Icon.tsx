import * as React from "react";

import { Icon as NativeBaseIcon, NativeBase } from "native-base";

export class Icon extends React.Component<NativeBase.Icon> {
  render() {
    return (
      <NativeBaseIcon {...this.props} />
    );
  }
}

export default Icon;
