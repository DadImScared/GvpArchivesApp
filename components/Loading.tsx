import * as React from "react";

import { Spinner } from "native-base";
import { View } from "react-native";

export const Loading: React.FunctionComponent<{ isLoading: boolean }> = ({ isLoading }) => isLoading ? (
  <View>
    <Spinner />
  </View>
) : null;

export default Loading;
