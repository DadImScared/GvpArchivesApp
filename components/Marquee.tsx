import * as React from "react";

import Ticker, { ITextMarqueeProps } from "react-native-text-ticker";

const ForwardedMarquee = React.forwardRef<Ticker, ITextMarqueeProps>(({ children, ...other }, ref) => (
  <Ticker {...other} ref={ref}>
    {children}
  </Ticker>
));

export default ForwardedMarquee;
