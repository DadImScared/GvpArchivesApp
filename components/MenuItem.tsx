import * as React from "react";

import { ListItem, Text } from "native-base";

import { IRoute } from "./SideBar";

export interface IMenuItemProps {
  to: string;
  display: IRoute;
  navigateRoute: (to: string) => void;
  active: boolean;
}

export class MenuItem extends React.Component<IMenuItemProps> {

  navigateRoute = () => {
    const { to, navigateRoute } = this.props;
    navigateRoute(to);

  };

  shouldComponentUpdate(nextProps: IMenuItemProps) {
    const { active } = this.props;
    return active !== nextProps.active;
  }

  render() {
    const { display, active } = this.props;
    return (
      <ListItem selected={active} onPress={this.navigateRoute} button={true}><Text>{display}</Text></ListItem>
    );
  }
}

export default MenuItem;
