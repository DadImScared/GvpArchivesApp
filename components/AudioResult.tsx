import * as React from "react";

import { Button, Card, CardItem, Left } from "native-base";
import { StyleSheet, Text } from "react-native";

import { Item } from "../reducers/itemsById";
import CardHeader from "./CardHeader";

export interface IAudioResultProps {
  item: Item;
  [key: string]: any;
}

const styles: any = StyleSheet.create({
  button: {
    alignItems: "center",
    justifyContent: "center",
    margin: 4,
    width: "30%"
  }
});

export const AudioResult: React.FC<IAudioResultProps> = ({ item }) => (
  <Card>
    <CardHeader item={item} />
    <CardItem footer={true}>
      <Left>
        <Button primary={true} style={styles.button}><Text style={{ color: "white" }}>Play</Text></Button>
        <Button info={true} style={styles.button}><Text>Playlist</Text></Button>
      </Left>
    </CardItem>
  </Card>
);

export default AudioResult;
