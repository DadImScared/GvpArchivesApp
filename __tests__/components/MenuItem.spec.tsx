import * as React from "react";
import * as renderer from "react-test-renderer";

import { IMenuItemProps, MenuItem } from "../../components/MenuItem";

let props: IMenuItemProps;

describe("MenuItem", () => {
  beforeEach(() => {
    props = {
      active: false,
      display: "Books",
      navigateRoute: jest.fn(),
      to: "book",
    };
  });

  it("should render", () => {
    const tree = renderer.create(<MenuItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
