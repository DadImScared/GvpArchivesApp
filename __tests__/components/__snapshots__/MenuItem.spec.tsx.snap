// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should render 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    button={true}
    onPress={[Function]}
    selected={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 0.5,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 16,
        "paddingBottom": 13,
        "paddingRight": 16,
        "paddingTop": 13,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      Books
    </Text>
  </View>
</View>
`;
